File: pytools/examples/csharp/client/Protocol/MsgTestXX.cs
using System.Collections;
using System.Collections.Generic;


public class MsgTestXX
{
	private byte _id_u8;
	private List<float> _id_f32 = new List<float>();
	private byte id_op_u8_flag;
	private byte _id_op_u8;


	public Packet Encode()
	{
		Packet packet = new Packet();
		packet.WriteByte(this._id_u8);
		ushort id_f32_count = (ushort)this._id_f32.Count;
		packet.WriteUshort(id_f32_count);
		for (ushort i = 0; i < id_f32_count; i++)
		{
			float xxx = this._id_f32[i];
			packet.WriteFloat(xxx);
		}
		packet.WriteByte(id_op_u8_flag);
		if (this.id_op_u8_flag == 1)
		{
			packet.WriteByte(this._id_op_u8);
		}
		return packet;
	}

	public MsgTestXX()
	{
	}

	public MsgTestXX(Packet packet)
	{
		this._id_u8 = packet.ReadByte();
		this._id_f32 = new List<float>();
		ushort id_f32_count = packet.ReadUshort();
		for (ushort i = 0; i < id_f32_count; i++)
		{
			this._id_f32.Add(packet.ReadFloat());
		}
		this. id_op_u8_flag = packet.ReadByte();
		if (this.id_op_u8_flag == 1)
		{
			this._id_op_u8 = packet.ReadByte();
		}
	}

	public List<byte> GetBuffer()
	{
		return this.Encode().GetBuffer();
	}


	public byte id_u8
	{
		get { return this._id_u8; }
		set { this._id_u8 = value; }
	}

	public List<float> id_f32
	{
		get { return this._id_f32; }
		set { this._id_f32 = value; }
	}

	public byte id_op_u8
	{
		get { return this._id_op_u8; }
		set { this.id_op_u8_flag = 1; this._id_op_u8 = value; }
	}

}

File: pytools/examples/java/client/protocol/MsgTestXX.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class MsgTestXX
{
	private byte id_u8;
	private List<Float> id_f32 = new ArrayList<Float>();
	private byte id_op_u8_flag;
	private byte id_op_u8;


	public MsgTestXX()
	{

	}

	public MsgTestXX(Packet packet)
	{
		id_u8 = packet.readByte();
		int id_f32_count = PacketUtil.readUShort(packet.readShort());
		for (int i = 0; i < id_f32_count; i++)
		{
			id_f32.add(packet.readFloat());
		}
		id_op_u8_flag = packet.readByte();
		if (id_op_u8_flag == 1)
		{
			id_op_u8 = packet.readByte();
		}
	}

	public byte[] encode()
	{
		Packet packet = new Packet();
		packet.writeByte(id_u8);
		int id_f32_count = id_f32.size();
		packet.writeShort((short)id_f32_count);
		for (int i = 0; i < id_f32_count; i++)
		{
			packet.writeFloat((float)id_f32.get(i));
		}
		packet.writeByte(id_op_u8_flag);
		if (id_op_u8_flag == 1)
		{
			packet.writeByte(id_op_u8);
		}
		return packet.readBytes();
	}

	public byte[] getBytes()
	{
		return this.encode();
	}



	public void setIdU8(byte id_u8)
	{
		this.id_u8 = id_u8;
	}
	public int getIdU8()
	{
		return PacketUtil.readUByte(this.id_u8);
	}

	public void setIdF32(List<Float> id_f32)
	{
		this.id_f32 = id_f32;
	}
	public List<Float> getIdF32()
	{
		return this.id_f32;
	}

	public void setIdOpU8(byte id_op_u8)
	{
		this.id_op_u8_flag = (byte)1;
		this.id_op_u8 = id_op_u8;
	}
	public int getIdOpU8()
	{
		return PacketUtil.readUByte(this.id_op_u8);
	}

}
